web-client/src/components/DashboardLayout/DashboardLayout.tsx

```diff
@@ -0,0 +1,49 @@
+import { Layout } from 'antd';
+import React, { useState } from 'react';
+import { MenuItem } from 'src/types/menu-item';
+
+import BottomNavbar from '../BottomNavbar/BottomNavbar';
+import SideDrawer from '../SideDrawer/SideDrawer';
+import TopNavbar from '../TopNavbar/TopNavbar';
+
+const DashboardLayout: React.FC<DashboardLayoutProps> = ({
+  menuItems,
+  profileData,
+  siteLocations,
+  children,
+}) => {
+  const [siderCollapsed, setSiderCollapsed] = useState(true);
+  return (
+    <Layout>
+      <TopNavbar />
+      <SideDrawer
+        collapsed={siderCollapsed}
+        closeSider={() => setSiderCollapsed(true)}
+        menuItems={menuItems}
+        profileData={profileData}
+        siteLocations={siteLocations}
+      />
+      <Layout.Content style={{ marginTop: '64px', marginBottom: '64px' }}>
+        {children}
+      </Layout.Content>
+      <BottomNavbar
+        openSider={() => setSiderCollapsed(false)}
+        siteLocations={siteLocations}
+      />
+    </Layout>
+  );
+};
+
+interface DashboardLayoutProps {
+  menuItems: Array<MenuItem>;
+  profileData: {
+    fullName: string;
+    photoUrl: string;
+    followers: number;
+    stars: number;
+  };
+  siteLocations: any;
+  children?: React.ReactNode;
```

Comment:
This is like a duplicate code. But, I really don't know how to fix it. Any advice? Can Context API solve it?

Reply:
Since this is an interface declaration I'd say it should be in a more global file, so you don't have to repeat it. It should then be available as a type in other files

Reply:
Refactored in a82010c